lib/x86_cpu/x86_env.ml

```diff
@@ -160,6 +162,61 @@ module R32 = struct
   (* Only 8 YMM/XMM registers in x86 32-bit mode *)
   let ymms = Array.sub ymms 0 8
 
+  (* mapping registers to arch variables *)
+  let of_reg reg =
+    let internal_reg = X86_asm_reg.decode reg in
+    match internal_reg with
+    | None -> None
+    | Some reg ->
+      (match reg with
+       | `AL | `AH | `AX | `EAX -> Some rax
```

Comment:
shouldn't we map to `exp` instead and preserve the width of the register? With such representation, we have that `bitwidth reg <> bitwidth var`.

Reply:
Basically, we can have `to_var : reg -> var option` that will be a one-to-one match, e.g., 
```ocaml
to_var `AL = None
to_var `EAX = if arch = `x86 then Some rax else None
to_var `RAX = if arch = `x86_64 then Some rax else None
```

and we can also have a `to_base_var` function, that will for the given register return its base vairant, e.g., RAX for AL in x86-64 mode.

Reply:
This would be equivalent to the RR.get function from x86_tools. Variables are sometimes a bit more convenient to work with compared to exp. In any case, I'll rename to `to_base_var` to avoid confusion. I'll see if I can move the RR.get functionality in here as well. Let me know what you think.

Reply:
sounds good, as long as we make it explicit and giving a user the options to choose from.